app/common/components/table.test.js

```diff
@@ -70,34 +71,57 @@ describe('table', () => {
     harness.request(context, ($) => {
       context.params.data.forEach((row, rowIndex) => {
         row.forEach((dataPoint, i) => {
-          expect($(`[data-test-id="table-row-${rowIndex}"] a:nth-child(${i + 1})`).text().trim()).toEqual(dataPoint);
-          expect($(`[data-test-id="table-row-${rowIndex}"] a:nth-child(${i + 1})`).hasClass(mockContext.params.columnClass)).toEqual(true);
-          expect($(`[data-test-id="table-row-${rowIndex}"] a:nth-child(${i + 1})`).attr('href')).toEqual('#');
+          const aTag = $(`[data-test-id="table-row-${rowIndex}"] a:nth-child(${i + 1})`);
+          expect(aTag.text().trim()).toEqual(dataPoint.data);
+          expect(aTag.hasClass(mockContext.params.columnClass)).toEqual(true);
+          expect(aTag.attr('href')).toEqual(dataPoint.href);
           expect($(`[data-test-id="table-row-${rowIndex}"] div`).length).toEqual(0);
         });
       });
     });
   }));
 
-  it('should render <div> for columns with link false property', createTestHarness(setup, (harness) => {
+  it('should render <div> for data with no link property', createTestHarness(setup, (harness) => {
     const context = { params: { ...mockContext.params } };
-    context.params.columnInfo[0].link = false;
-    context.params.columnInfo[1].link = false;
+    delete context.params.data[0][0].href;
+    delete context.params.data[0][1].href;
+    delete context.params.data[1][0].href;
+    delete context.params.data[1][1].href;
     harness.request(context, ($) => {
       context.params.data.forEach((row, rowIndex) => {
         row.forEach((dataPoint, i) => {
-          expect($(`[data-test-id="table-row-${rowIndex}"] div:nth-child(${i + 1})`).text().trim()).toEqual(dataPoint);
-          expect($(`[data-test-id="table-row-${rowIndex}"] div:nth-child(${i + 1})`).hasClass(mockContext.params.columnClass)).toEqual(true);
+          const divTag = $(`[data-test-id="table-row-${rowIndex}"] div:nth-child(${i + 1})`);
+          expect(divTag.text().trim()).toEqual(dataPoint.data);
+          expect(divTag.hasClass(mockContext.params.columnClass)).toEqual(true);
           expect($(`[data-test-id="table-row-${rowIndex}"] a`).length).toEqual(0);
         });
       });
     });
   }));
+
+  it('should render tag component for columns with tag property', createTestHarness(setup, (harness) => {
+    const context = { params: { ...mockContext.params } };
+    delete context.params.data[0][0].href;
+    delete context.params.data[0][1].href;
+    delete context.params.data[1][0].href;
+    delete context.params.data[1][1].href;
+    context.params.data[0][0].tag = {
+      dataTestId: 'a-tag-id-1',
+      text: 'tag text',
+      classes: 'a-class',
+    };
+    harness.request(context, ($) => {
+      expect($('[data-test-id="a-tag-id-1"]').text().trim()).toEqual(context.params.data[0][0].tag.text);
+      expect($('[data-test-id="a-tag-id-1"]').hasClass(context.params.data[0][0].tag.classes)).toEqual(true);
+    });
+  }));
```

Comment:
A general comment do you think it might be clearer to provide 3 examples of contexts. One with link, one without link and one with a tag. It'll just be bit more obvious the context that is provided to these templates and what will get rendered